### PR TITLE
Added PHP 8.0 features

### PR DESCRIPTION
**Pull request by TomPavelec**
feat: named arguments
feat: added nullsafe operator, bitwise operators
feat: attributes

### CODE REVIEW

File: components/prism-php.js [hunk 1]
```diff
@@ -139,10 +158,42 @@
...
 
+	Prism.languages.insertBefore('php', 'comment', {
+		'attribute': {
+			pattern: /#\[(?:(?!#\[)(?:\\[\s\S]|[^\\\]]))*]/,
+			greedy: true,
+			inside: {
+				'interpolation': {
+					pattern: /\[(?:(?!#\[)(?:\\[\s\S]|[^\\\]]))*]/,
+					inside: {
+						rest: Prism.languages.php,
```
Comment (RunDevelopment) [inline components/prism-php.js:173]: Please move `rest` below `attribute-class-name`. `rest` will cause the token of `Prism.languages.php` to be inserted at the end of the `inside` object. Having it here is misleading.

**Review by RunDevelopment — CHANGES_REQUESTED** [review]
Thank you @TomPavelec for another amazing PR!

File: components/prism-php.js (hunk 1 above)
Comment (TomPavelec) [reply components/prism-php.js]: fixed

File: components/prism-php.js [hunk 2]
```diff
@@ -139,10 +158,42 @@
...
 		},
 		'number': /\b0b[01]+\b|\b0x[\da-f]+\b|(?:\b\d+(?:_\d+)*\.?(?:\d+(?:_\d+)*)*|\B\.\d+)(?:e[+-]?\d+)?/i,
-		'operator': /<?=>|\?\?=?|\.{3}|->|[!=]=?=?|::|\*\*=?|--|\+\+|&&|\|\||[?~]|[/^|%*&<>.+-]=?/,
+		'operator': /<?=>|\?\?=?|\.{3}|\??->|[!=]=?=?|::|\*\*=?|--|\+\+|&&|\|\||<<|>>|[?~]|[/^|%*&<>.+-]=?/,
 		'punctuation': /[{}\[\](),:;]/
 	};
 
+	Prism.languages.insertBefore('php', 'comment', {
+		'attribute': {
+			pattern: /#\[(?:(?!#\[)(?:\\[\s\S]|[^\\\]]))*]/,
```
[the remark this answers is not in the document]
Comment (TomPavelec) [reply components/prism-php.js]: I added some cases to tests and improved attribute pattern.
But i think there are still some edge cases.
(nested attrs definitely will not work with this)

File: components/prism-php.js [hunk 3]
```diff
@@ -139,10 +158,42 @@
...
+			greedy: true,
+			inside: {
+				'interpolation': {
```
[the remark this answers is not in the document]
Comment (RunDevelopment) [reply components/prism-php.js]: Yes. (Don't forget a `punctuation` alias or there won't be any highlighting.)

File: components/prism-php.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (RunDevelopment) [reply components/prism-php.js]: > nested attrs definitely will not work with this

Nested attributes aren't supported by PHP yet, I believe, so there's no reason to support them at this point in time.

> But i think there are still some edge cases.

How about this one: `#[Foo("]")]`. This is why I suggested `/#\[(?:["'\[\]]|<php-strings>)*\]/`. 
(Just noticed that I forgot to escape the outer brackets...)

Also, comments.

```php
#[
  // something ]
  Foo,
  /* something else #[] */
  Bar
  # shell comments aren't confusing at all in here
]
```

The full attribute pattern (without nested attributes) is:

```
/#\[(?:[^"'/#\[\]]|\/(?![*/])|("|')(?:\\[\s\S]|(?!\1)[^\\])*\1|(?:\/\/|#(?!\[)).*$|\/\*(?:[^*]|\*(?!\/))*\*\/)*\]/m
```

Easy, right... The whole this is basically a modified `phpPattern`.

[the remark this answers is not in the document]
Comment (TomPavelec) [reply components/prism-php.js]: This is mindblowing :smile:. This pattern is too complex for me. It properly matches comments in attrs. But do not match array `[...]`.
e.g. `#[Attr(3, [2, 3], 'str')]` is not matched.
I can't fix it, it's beyond me.

[the remark this answers is not in the document]
Comment (RunDevelopment) [reply components/prism-php.js]: I didn't know you could do `[1,2]` arrays in PHP... This makes things more complicated because we have to count the number of opening and closing brackets which isn't possible with regular expressions. The best we can do is to support a maximum number of nested brackets (e.g. 2 or 3 levels) and call it a day. The resulting pattern will be really long and repetitive, so here is a function that generates it:

```js
function () {
	// <phpExpression>      ::= (?:[^"'/#\[\]]|<division>|<string>|<comment>|#?\[<phpExpression>\])*
	// <string>             ::= "(?:\\[\s\S]|[^\\"])*"|'(?:\\[\s\S]|[^\\'])*'
	// <comment>            ::= <singleline-comment>|<multiline-comment>
	// <singleline-comment> ::= (?:\/\/|#(?!\[)).*$  ; m flag is enabled
	// <multiline-comment>  ::= \/\*(?:[^*]|\*(?!\/))*\*\/
	// <division>           ::= \/(?![*/])  ; this is necessary to match all "/" chars that don't start a comment
	var phpExpression = /(?:[^"'/#\[\]]|\/(?![*/])|"(?:\\[\s\S]|[^\\"])*"|'(?:\\[\s\S]|[^\\'])*'|(?:\/\/|#(?!\[)).*$|\/\*(?:[^*]|\*(?!\/))*\*\/|#?\[<SELF>\])*/.source;
	
	// This will add support for 4 levels of nested [] brackets.
	for (var i = 0; i < 2; i++) {
		phpExpression = phpExpression.replace(/<SELF>/g, function () { return phpExpression; });
	}
	phpExpression = phpExpression.replace(/<SELF>/g, function () { return /[^\s\S]/.source; });
	
	return RegExp('#\\[' + phpExpression + '\\]', 'm');
}
```

Since we now support nested `[]` brackets, I added support for nested attributes as well because it only takes a `#?` before the nested `\[<SELF>\]`-part to do it.

File: components/prism-php.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (TomPavelec) [reply components/prism-php.js]: `delimiter` with `punctuation` alias added

File: components/prism-php.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (TomPavelec) [reply components/prism-php.js]: Thank you very much for commented function code. I realized that attributes and arrays can appear in different contexts, so I can match attribute with nested array inside without pattern recursion.

[the remark this answers is not in the document]
Comment (RunDevelopment) [reply components/prism-php.js]: Your solution is really nice. Let's go with it!

**Comment by TomPavelec** [comment]
I forgot union types, just wait a minute with merge, please.

File: components/prism-php.js [hunk 4]
```diff
@@ -149,64 +172,112 @@
...
+		}
+	];
+
+	Prism.languages.insertBefore('php', 'variable', {
+		'string': string,
+	});
+
+	Prism.languages.insertBefore('php', 'variable', {
+		'attribute': {
+			pattern: /#\[(?:(?!#\[)(?:[^"'\/#]|\/\/.*$|#.*$|\/\*(?:[^*]|\*(?!\/))*\*\/|"(?:\\[\s\S]|[^\\"])*"|'(?:\\[\s\S]|[^\\'])*'))+](?=\s*[a-z$#])/mi,
```
Comment (RunDevelopment) [inline components/prism-php.js:257]: 1. The alternative for division `\/(?![*/])` was missing.
1. `(?!#\[)` can be moved inside the alternation. It only affects the `#.*$` alternative, so that's the only one that has to be changed.

Also, please always escape `]` characters.

```suggestion
			pattern: /#\[(?:[^"'\/#]|\/(?![*/])|\/\/.*$|#(?!\[).*$|\/\*(?:[^*]|\*(?!\/))*\*\/|"(?:\\[\s\S]|[^\\"])*"|'(?:\\[\s\S]|[^\\'])*')+\](?=\s*[a-z$#])/mi,
```

File: tests/languages/php/attribute_feature.test [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+function foo(
+    #[Attribute] $param1,
+    $param2
+) {}
+
+$f1 = #[ExampleAttribute] function () {};
+
+$ref = new \ReflectionFunction(#[A1] #[A2] function () { });
+
+#[DeprecationReason('reason: <https://some-website/reason?>')]
```
Comment (RunDevelopment) [inline tests/languages/php/attribute_feature.test:54]: Without an opening `<?php` tag, this test isn't very interesting. The intention behind this was to trip `phpPattern`. Please make a new test file with the following code:

```php
<?php
#[DeprecationReason('reason: <https://some-website/reason?>')]
function main() {}
?>
```

Comment (TomPavelec) [reply tests/languages/php/attribute_feature.test:54]: Test case added.

**Review by RunDevelopment — APPROVED** [review]

**Comment by RunDevelopment** [comment]
Thank you very much for contributing @TomPavelec!

**Comment by TomPavelec** [comment]
> Thank you very much for contributing @TomPavelec!

I thank you, for your support and advice, I couldn't have done it alone.
